src/Components/CreateImageWizardV2/ImageWizardContext.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { useState } from 'react';
+
+import {
+  EnvironmentStateType,
+  filterEnvironment,
+  useGetAllowedTargets,
+} from './steps/ImageOutput/Environment';
+import { GCPAccountTypes } from './steps/TargetEnvironment/GCP/GCPTarget';
+import { useGetAccountData } from './steps/TargetEnvironment/SourcesSelect';
+
+import { RHEL_9, X86_64 } from '../../constants';
+import { ArchitectureItem, Distributions } from '../../store/imageBuilderApi';
+
+type ImageWizardContextPropType = {
+  releaseState: [
+    Distributions,
+    React.Dispatch<React.SetStateAction<Distributions>>
+  ];
+  architectureState: [
+    ArchitectureItem['arch'],
+    React.Dispatch<React.SetStateAction<ArchitectureItem['arch']>>
+  ];
+  environmentState: [
+    EnvironmentStateType,
+    React.Dispatch<React.SetStateAction<EnvironmentStateType>>
+  ];
+  isAwsManualState: [boolean, React.Dispatch<React.SetStateAction<boolean>>];
+  awsSourceState: [
+    [number, string],
+    React.Dispatch<React.SetStateAction<[number, string]>>
+  ];
+  associatedAwsAccountIdState: [
+    string,
+    React.Dispatch<React.SetStateAction<string>>
+  ];
+  isShareGoogleAccountState: [
+    boolean,
+    React.Dispatch<React.SetStateAction<boolean>>
+  ];
+  gcpAccountTypeState: [
+    GCPAccountTypes,
+    React.Dispatch<React.SetStateAction<GCPAccountTypes>>
+  ];
+  gcpAccountEmailState: [string, React.Dispatch<React.SetStateAction<string>>];
+  gcpDomainState: [string, React.Dispatch<React.SetStateAction<string>>];
+  isAzureManualState: [boolean, React.Dispatch<React.SetStateAction<boolean>>];
+  azureSourceState: [
+    [number, string],
+    React.Dispatch<React.SetStateAction<[number, string]>>
+  ];
+  azureTenantIdState: [string, React.Dispatch<React.SetStateAction<string>>];
+  azureSubscriptionIdState: [
+    string,
+    React.Dispatch<React.SetStateAction<string>>
+  ];
+  azureResourceGroupState: [
+    string,
+    React.Dispatch<React.SetStateAction<string>>
+  ];
+};
+
+export const ImageWizardContext =
+  React.createContext<ImageWizardContextPropType>({
+    releaseState: [RHEL_9, () => {}],
+    architectureState: [X86_64, () => {}],
+    environmentState: [
+      {
+        aws: { selected: false, authorized: false },
+        azure: { selected: false, authorized: false },
+        gcp: { selected: false, authorized: false },
+        oci: { selected: false, authorized: false },
+        'vsphere-ova': { selected: false, authorized: false },
+        vsphere: { selected: false, authorized: false },
+        'guest-image': { selected: false, authorized: false },
+        'image-installer': { selected: false, authorized: false },
+        wsl: { selected: false, authorized: false },
+      },
+      () => {},
+    ],
+    isAwsManualState: [false, () => {}],
+    awsSourceState: [[0, ''], () => {}],
+    associatedAwsAccountIdState: ['', () => {}],
```

Comment:
Do we want to initialize things to empty strings? Wouldn't `undefined` be more accurate? In the type, you can use a union of `string | undefined`.

Reply:
That's a big question, having undefined means we'll have to deal with protection everywhere no?

Reply:
It's a very typical pattern in our code base. We check for whether something is undefined or not using optional chaining (`thing?.otherthing`) and can take advantage of short circuiting if it is undefined (`thing?.otherthing && ...`). Our linter will ensure that we never attempt to access something that is undefined.

Reply:
Yeah, atm the empty string only needs to be checked to validate the step, which makes it convenient for everywhere else it needs to be used. But as you prefer!